lib/dynosaur/inputs/new_relic_input_plugin.rb

```diff
@@ -0,0 +1,63 @@
+require 'dynosaur/new_relic_api_client'
+require 'dynosaur/version'
+require 'dynosaur/error_handler'
+
+# ScalerPlugin implementation that uses the New Relic API
+# to get the current requests per minute on the site, and scale to an
+# appropriate number of dynos.
+
+module Dynosaur
+  module Inputs
+    class NewRelicInputPlugin < AbstractInputPlugin
+
+      # Load config from the config hash
+      def initialize(config)
+        super
+
+        @unit = "RPM"
+        @key = config["key"]
+        @app_id = config["app_id"].to_s
+        @rpm_per_dyno = config["rpm_per_dyno"].to_i
+
+        if @key.blank?
+          raise "You must supply API key in the new relic plugin config"
+        end
+        if @app_id.blank?
+          raise "You must supply appid in the new relic plugin config"
+        end
+
+        @metric_name = "HttpDispatcher"
+        @new_relic_api_client = Dynosaur::NewRelicApiClient.new(@key, @app_id)
+      end
+
+      def retrieve
+        return get_rpm
+      end
+
+      def value_to_resources(value)
+        estimate = (value / @rpm_per_dyno.to_f).ceil
+        puts "NEW RELIC: #{@recent} => #{value} : #{estimate}"
+        return estimate
+      end
+
+      private
+
+      def get_rpm
+        begin
+          # New Relic is not very accurate and sometimes returns data for the
+          # current minute, as if it was already complete.
+          # This is not an issue here because we always use the max value from
+          # the ring buffer. This would become an issue if we had an
+          # hysteresis_period smaller than 1 minute.
```

Comment:
we might want to raise an exception, or maybe issue a warning if the hysteresis_period is shorter than a minute